common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -835,6 +835,24 @@ public int indexOf(UTF8String v, int start) {
     return -1;
   }
 
+  public int indexOf(UTF8String substring, int start, int collationId) {
```

Comment:
I think we now have a lot of new methods 
consider adding comments such as: https://github.com/apache/spark/pull/45791/files#r1549415554